### PR TITLE
improve loader

### LINKED ISSUE #4: Improve loader component
Add variation to loader

### LINKED ISSUE #6: Add definition

### PR DESCRIPTION
**Pull request by pawan-blotout**
Resolves #4 
Resolves #6

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
+interface DropdownValue {
```
Comment (NejcZdovc) [inline src/index.d.ts:1]: we need to define which props are optional and which are required

Comment (pawan-blotout) [reply src/index.d.ts:1]: Defined optional parameters

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  href?: string
+  icon?: React.ReactChild | React.ReactChild[]
+  children?: React.ReactChild | React.ReactChild[]
+  onClick: () => void
+}
+
+interface DataTableInterface {
+  rows: string[][]
+  noData?: string[]
+  headers: string[]
```
Comment (NejcZdovc) [inline src/index.d.ts:44]: this should be optional

File: src/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  iconPosition?: string
+  to?: string
+  href?: string
+  icon?: React.ReactChild | React.ReactChild[]
+  children?: React.ReactChild | React.ReactChild[]
+  onClick: () => void
+}
+
+interface DataTableInterface {
+  rows: string[][]
```
Comment (NejcZdovc) [inline src/index.d.ts:42]: this is string | React.ReactElement

File: src/index.d.ts (hunk 2 above)
Comment (NejcZdovc) [inline src/index.d.ts:44]: this is string | React.ReactElement

File: src/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    label: string
+    value: string
+    number: number
+  }
+  loadOptions: (arg0: string) => {
+    label: string
+    value: number
+  }
+  noOptionsMessage?: (arg0: string) => string
+  isClearable: boolean
```
Comment (NejcZdovc) [inline src/index.d.ts:78]: this should be optional

File: src/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  sideHeading: string
```
Comment (NejcZdovc) [inline src/index.d.ts:79]: this should be optional

File: src/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+interface LineInterface {
+  id?: string
+  data: {
+    bars: string[]
+    categories: string[]
+    data: [
+      {
+        data: number[]
+        breakDown: Record<string, number>[]
```
Comment (NejcZdovc) [inline src/index.d.ts:107]: this should be optional

File: src/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+interface MultiSelectInputInterface {
+  minHeight?: number
+  error?: boolean
+  className?: string
+  inputValue: string | number
+  isClearable?: boolean
+  isMulti?: boolean
+  handleChange: () => void
+  handleInputChange: () => void
```
Comment (NejcZdovc) [inline src/index.d.ts:121]: this should be optional

File: src/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+  handleKeyDown: () => void
```
Comment (NejcZdovc) [inline src/index.d.ts:122]: this should be optional

File: src/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+interface SimpleSelectInterface {
+  value: DropdownValue
+  showError?: boolean
+  type?: string
+  styles?: Record<string, number | string>[]
+  isClearable?: boolean
+  isMulti?: boolean
+  handleChange: () => void
+  handleInputChange: () => void
```
Comment (NejcZdovc) [inline src/index.d.ts:135]: this should be optional

File: src/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+  handleKeyDown: () => void
```
Comment (NejcZdovc) [inline src/index.d.ts:136]: this should be optional

File: src/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+  variant?: string
+  message: string
+}
+
+interface SpinnerInterface {
+  type?: string
+}
+
+interface TableInterface {
+  headers: string[]
```
Comment (NejcZdovc) [inline src/index.d.ts:169]: this should be optional

File: src/index.d.ts [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+  format?: (arg0: number, arg1: number, arg2: string) => string
+  noData?: string[]
+}
+
+interface TextareaInterface {
+  className?: string
+  error?: boolean
+}
+
+interface ToggleInterface {
```
Comment (NejcZdovc) [inline src/index.d.ts:181]: this props are not correct

File: src/index.d.ts [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+
+interface TableInterface {
+  headers: string[]
+  rows: string[][]
+  className?: string
+  format?: (arg0: number, arg1: number, arg2: string) => string
+  noData?: string[]
+}
+
+interface TextareaInterface {
```
Comment (NejcZdovc) [inline src/index.d.ts:176]: not all props are listed here

File: src/index.d.ts [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+}
+
+interface SnackbarInterface {
+  onClose: () => void
+  variant?: string
+  message: string
+}
+
+interface SpinnerInterface {
+  type?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:165]: we should define which types are possible

File: src/index.d.ts [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+  isDisabled?: boolean
+  isMulti?: boolean
+  isClearable?: boolean
+  noOptionsMessage?: (arg0: string) => string
+  placeholder?: string
+}
+
+interface SnackbarInterface {
+  onClose: () => void
+  variant?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:160]: which variants are possible

File: src/index.d.ts [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+  handleChange: () => void
+  handleInputChange: () => void
+  handleKeyDown: () => void
+  placeholder?: string
+}
+
+interface SimpleAsyncSelectInterface {
+  value: DropdownValue
+  showError?: boolean
+  type?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:143]: which types are possible

File: src/index.d.ts [hunk 17]
```diff
@@ -0,0 +1,209 @@
...
+  handleInputChange: () => void
+  handleKeyDown: () => void
+  placeholder?: string
+  value: string[]
+}
+
+interface SimpleSelectInterface {
+  value: DropdownValue
+  showError?: boolean
+  type?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:130]: which types are possible

File: src/index.d.ts [hunk 18]
```diff
@@ -0,0 +1,209 @@
...
+interface FieldWrapperInterface {
+  className?: string
+  html?: boolean
+  children: React.ReactChild | React.ReactChild[]
+  helperType?: string
+  helperText?: string
+  label?: string
+}
+
+interface InputInterface {
```
Comment (NejcZdovc) [inline src/index.d.ts:91]: some props are missing

File: src/index.d.ts [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+interface DialogInterface {
+  className?: string
+  title: string
+  children: React.ReactChild | React.ReactChild[]
+  onClose: () => void
+  variant?: string
+  disabledOutline?: boolean
+}
+
+interface DropdownInterface {
```
Comment (NejcZdovc) [inline src/index.d.ts:65]: some props are missing

File: src/index.d.ts [hunk 20]
```diff
@@ -0,0 +1,209 @@
...
+  searchable?: boolean
+  hasPagination?: boolean
+}
+
+interface DialogInterface {
+  className?: string
+  title: string
+  children: React.ReactChild | React.ReactChild[]
+  onClose: () => void
+  variant?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:61]: which variants are possible

File: src/index.d.ts [hunk 21]
```diff
@@ -0,0 +1,209 @@
...
+
+interface DataTableInterface {
+  rows: string[][]
+  noData?: string[]
+  headers: string[]
+  emptyCell?: string
+  className?: string
+  Component?: React.ReactChild | React.ReactChild[]
+  loader?: boolean
+  type?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:49]: which types are possible

File: src/index.d.ts [hunk 22]
```diff
@@ -0,0 +1,209 @@
...
+
+interface BoxInterface {
+  className?: string
+  type?: string
+  loading?: boolean
+}
+
+interface ButtonInterface {
+  className?: string
+  type?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:28]: which types are possible

File: src/index.d.ts [hunk 23]
```diff
@@ -0,0 +1,209 @@
...
+  variant?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:29]: which variants are possible

File: src/index.d.ts [hunk 24]
```diff
@@ -0,0 +1,209 @@
...
+  size?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:30]: which sizes are possible

File: src/index.d.ts [hunk 25]
```diff
@@ -0,0 +1,209 @@
...
+  isDisabled?: boolean
+  id?: string
+  iconPosition?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:33]: which iconPositions are possible

File: src/index.d.ts [hunk 26]
```diff
@@ -0,0 +1,209 @@
...
+  vertical?: boolean
+  labelFormatter?: (label: string) => string
+  percent?: boolean
+  showLegend?: boolean
+  id?: string
+}
+
+interface BoxInterface {
+  className?: string
+  type?: string
```
Comment (NejcZdovc) [inline src/index.d.ts:22]: which types are possible

File: src/index.d.ts (hunk 2 above)
Comment (pawan-blotout) [reply src/index.d.ts]: Done

File: src/index.d.ts (hunk 3 above)
Comment (pawan-blotout) [reply src/index.d.ts]: Done

File: src/index.d.ts (hunk 2 above)
Comment (pawan-blotout) [reply src/index.d.ts]: Done

File: src/index.d.ts (hunk 4 above)
Comment (pawan-blotout) [reply src/index.d.ts]: Done

File: src/index.d.ts (hunk 5 above)
Comment (pawan-blotout) [reply src/index.d.ts]: Done

File: src/index.d.ts (hunk 6 above)
Comment (pawan-blotout) [reply src/index.d.ts]: Done

File: src/index.d.ts [hunk 27]
```diff
@@ -0,0 +1,225 @@
...
+  handleAdd?: () => void
+  position?: 'left'
+  isMulti?: boolean
+  handleChange: () => void
+  value: {
+    label: string
+    value: string
+    number: number
+  }
+  loadOptions: (arg0: string) => {
```
Comment (NejcZdovc) [inline src/index.d.ts:79]: should this be (arg0: string) => DropdownValue[]?

File: src/index.d.ts [hunk 28]
```diff
@@ -0,0 +1,225 @@
...
+  className?: string
+  html?: boolean
+  children: React.ReactChild | React.ReactChild[]
+  helperType?: string
+  helperText?: string
+  label?: string
+}
+
+interface InputInterface {
+  type: 'text'
```
Comment (NejcZdovc) [inline src/index.d.ts:103]: this can be a lot more, we can just use `HTMLInputTypeAttribute`. And this prop is optional

File: src/index.d.ts [hunk 29]
```diff
@@ -0,0 +1,225 @@
+interface DropdownValue {
```
Comment (NejcZdovc) [inline src/index.d.ts:1]: Can we change this to SelectValue? So that is more generic that Dropdown

File: src/index.d.ts [hunk 30]
```diff
@@ -0,0 +1,225 @@
...
+  isDisabled?: boolean
+  isMulti?: boolean
+  isClearable?: boolean
+  noOptionsMessage?: (arg0: string) => string
+  placeholder?: string
+}
+
+interface SnackbarInterface {
+  onClose: () => void
+  variant?: 'success'
```
Comment (NejcZdovc) [inline src/index.d.ts:173]: `'success' | 'error'`

File: src/toggle/icon.d.ts [hunk 31]
```diff
@@ -0,0 +1,70 @@
+interface IconInterface {
```
Comment (NejcZdovc) [inline src/toggle/icon.d.ts:1]: why is this in toggle folder? shouldn't be in inside icon folder? Also does icons work? as we only do index.jsx as entry

**Review by NejcZdovc — APPROVED** [review]

**Pull request merged**